packages/flutter_test/test/finders_test.dart

```diff
@@ -390,6 +395,600 @@ void main() {
+
+      const String customDescription = 'custom description';
+      late TestFailure failure;
+      try {
+        expect(find.byElementPredicate((_) => false, description: customDescription), findsOneWidget);
+      } on TestFailure catch (e) {
+        failure = e;
+      }
+
+      expect(failure, isNotNull);
+      expect(failure.message, contains('Actual: _ElementPredicateWidgetFinder:<Found 0 widgets with $customDescription'));
+    });
+  });
+
+  group('find.byWidgetPredicate', () {
+    testWidgets('fails with a custom description in the message', (WidgetTester tester) async {
+      await tester.pumpWidget(const Text('foo', textDirection: TextDirection.ltr));
+
+      const String customDescription = 'custom description';
+      late TestFailure failure;
+      try {
+        expect(find.byWidgetPredicate((_) => false, description: customDescription), findsOneWidget);
+      } on TestFailure catch (e) {
+        failure = e;
+      }
+
+      expect(failure, isNotNull);
+      expect(failure.message, contains('Actual: _WidgetPredicateWidgetFinder:<Found 0 widgets with $customDescription'));
+    });
+  });
+
+  group('find.descendant', () {
+    testWidgets('finds one descendant', (WidgetTester tester) async {
+      await tester.pumpWidget(const Row(
+        textDirection: TextDirection.ltr,
+        children: <Widget>[
+          Column(children: fooBarTexts),
+        ],
+      ));
+
+      expect(find.descendant(
+        of: find.widgetWithText(Row, 'foo'),
+        matching: find.text('bar'),
+      ), findsOneWidget);
+    });
+
+    testWidgets('finds two descendants with different ancestors', (WidgetTester tester) async {
+      await tester.pumpWidget(const Row(
+        textDirection: TextDirection.ltr,
+        children: <Widget>[
+          Column(children: fooBarTexts),
+          Column(children: fooBarTexts),
+        ],
+      ));
+
+      expect(find.descendant(
+        of: find.widgetWithText(Column, 'foo'),
+        matching: find.text('bar'),
+      ), findsNWidgets(2));
+    });
+
+    testWidgets('fails with a descriptive message', (WidgetTester tester) async {
+      await tester.pumpWidget(const Row(
+        textDirection: TextDirection.ltr,
+        children: <Widget>[
+          Column(children: <Text>[Text('foo', textDirection: TextDirection.ltr)]),
+          Text('bar', textDirection: TextDirection.ltr),
+        ],
+      ));
+
+      late TestFailure failure;
+      try {
+        expect(find.descendant(
+          of: find.widgetWithText(Column, 'foo'),
+          matching: find.text('bar'),
+        ), findsOneWidget);
+      } on TestFailure catch (e) {
+        failure = e;
+      }
+
+      expect(failure, isNotNull);
+      expect(
+        failure.message,
+        contains(
+          'Actual: _DescendantWidgetFinder:<Found 0 widgets with text "bar" descending from root widgets with type "Column" that are ancestors of leaf widgets with text "foo"',
```

Comment:
can we remove the term "root widget" and "leaf widget" from this message? I find those terms really confusing.

Could this be reworded to:

Found 0 widgets with text "bar" that are descending from a "Column" widget, which is an ancestor of a widget with text "foo".

or something similar.

Reply:
Root and leaf can probably be removed (the intent was to describe the ends of the search, not necessarily their place in the entire tree, but it sounds like that's not clear at a glance). But some of the issues with the awkwardness are a result of composing multiple descriptions, so the "with type "Column"" part is actually coming from a _different_ finder. The descendent and ancestor finders (Semantics and Widget ones) are particularly prone to awkward wording because of that...

Reply:
Removed.